pogo/util.py

```diff
@@ -13,10 +40,58 @@ def f2h(float):
 def h2f(hex):
     return struct.unpack('<d', struct.pack('<Q', int(hex, 16)))[0]
 
+def d2h(f):
+   hex = f2h(f)[2:].replace('L','')
+   hex = ("0" * (len(hex) % 2)) + hex
+   return  hex.decode("hex")
 
 def encodeLocation(loc):
     return (f2i(loc.latitude), f2i(loc.longitude), f2i(loc.altitude))
 
 
 def getMs():
     return int(round(time.time() * 1000))
+
+
+def hashLocation(authTicket, latitude, longitude, altitude):
+    baseHash = xxhash.xxh32(authTicket.SerializeToString(), seed=0x1B845238).intdigest()
+    locationBytes = d2h(latitude) + d2h(longitude) + d2h(altitude)
+
+    # Using serialized Auth Ticket
+    hashA = xxhash.xxh32(locationBytes, seed=baseHash).intdigest()
+
+    #Hash of location using static seed 0x1B845238
+    hashB = xxhash.xxh32(locationBytes, seed=0x1B845238).intdigest()
+    return hashA, hashB
+    
+
+def hashRequests(authTicket, payload):
+    baseHash = xxhash.xxh64(authTicket.SerializeToString(), seed=0x1B845238).intdigest()
+    return xxhash.xxh64(payload.SerializeToString(), seed=baseHash).intdigest() # for request in payload]
+
+
+# Assuming the encrypt.dll file floating around out there
+def hashSignature(signature, libraryPath):
+    serialized = signature.SerializeToString()
+    size = len(serialized)
+
+    library = ctypes.cdll.LoadLibrary(libraryPath)
```

Comment:
this is good for now but I guess we should look into loading from `LD_LIBRARY_PATH`